### PR TITLE
Ctulo2

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Shiwa is dirty (due to me).
So I reimplemented it.
I also modified a few things.
This also includes an improvement of cGA.
Other improvements are expected.
The plan is (i) merge this PR (ii) check performance (iii) if everything fine, remove Shiwa and rename Ctulo as NGO10 or something like that.

## How Has This Been Tested (if it applies)

CI and a few preliminary tests.
Additional tests have been added.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -1755,3 +1756,70 @@ def _internal_ask_candidate(self) -> p.Parameter:
 
     def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
         self._optim.tell(candidate, value)
+
+
+@registry.register
+class Ctulo(base.Optimizer):
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1762]: why don't you rename it directly NGO2?

**Comment by jrapin** [comment]
Please don't remove Shiwa so soon, now that it was advised to use it and published, it is not user-friendly to remove it.
When NGO2 is working, I would recommend to make Shiwa inherit from it and just add a deprecation warning to it. This way it will still works for some time, remove most of the code, and allow users to adapt.
Also, if there are more NGO versions coming, we need to make `NGO` point to the latest version. This way we can keep NGO2, NGO3 etc as long as they make sence, and advise people to always use NGO as it is the one supposed to have the best results.

**Comment by jrapin** [comment]
Also, in what is it cleaner that Shiwa? :s

File: nevergrad/optimization/optimizerlib.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py:1762]: Ok I do so.

Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py:1762]: Well NGO10 because we have had several variants with various numbers.

**Comment by teytaud** [comment]
Ok for a NGO pointing to NGO-last.
I have no plan to remove Shiwa in the near future, and ok I will not remove it at all actually given that there are already several publications mentioning it.

**Comment by teytaud** [comment]
> Also, in what is it cleaner that Shiwa? :s

Shiwa has dead branches.
It is also longer than necessary, with several equivalent branches.

**Pull request merged**